crates/sui-types/src/transaction.rs

```diff
@@ -2074,9 +2074,9 @@ impl SenderSignedData {
         let mut mapping = BTreeMap::new();
         for sig in &self.inner().tx_signatures {
             if verify_legacy_zklogin_address {
-                // Try deriving the address from the legacy way.
+                // Try deriving the address from the legacy padded way.
                 if let GenericSignature::ZkLoginAuthenticator(z) = sig {
-                    mapping.insert(SuiAddress::legacy_try_from(z)?, sig);
+                    mapping.insert(SuiAddress::try_from_padded(&z.inputs)?, sig);
```

Comment:
shouldn't this be unpadded?

Reply:
legacy == padded

on line 2082: `let address = sig.try_into()?;` refers to the following:

```
impl TryFrom<&ZkLoginAuthenticator> for SuiAddress {
    type Error = SuiError;
    fn try_from(authenticator: &ZkLoginAuthenticator) -> SuiResult<Self> {
        SuiAddress::try_from_unpadded(&authenticator.inputs)
    }
}
```

therefore on line 2079, it should use try_from_padded so both are included in the mapping